### PR TITLE
Feature/parmmg

### PR DESCRIPTION
**Pull request by lcirrottola**
Update to the latest parmmg develop + automatic download of a specific version of mmg for parmmg private usage + some parameters

### CODE REVIEW

File: 3rdparty/parmmg/Makefile [hunk 1]
```diff
@@ -33,10 +33,8 @@ FAIT.done:tag-tar
 	-DMPI_CXX_COMPILER="$(MPICXX)" \
 	-DMPI_Fortran_COMPILER="$(MPIFC)" \
 	-DMPIEXEC_EXECUTABLE="$(MPIRUN)" \
-	-DDOWNLOAD_MMG=OFF \
+	-DDOWNLOAD_MMG=ON \
```
Comment (prj-) [inline 3rdparty/parmmg/Makefile:36]: This is kind of problematic, because we already install Mmg, so users would have two libraries. Isn't it possible to bump the Mmg archive version in `getall` as well in order to track the same commit as you have in ParMmg?

Comment (lcirrottola) [reply 3rdparty/parmmg/Makefile:36]: Updating the Mmg commit certainly can be much cleaner and allow to load both the sequential and parallel remeshers (altough users should be aware that the version of mmg used by parmmg is well ahead mmg/master, and some commit behind mmg/develop)

Comment (lcirrottola) [reply 3rdparty/parmmg/Makefile:36]: Maybe setting mmg include and ldflags to point inside 3rdparty/parmmg/build/Mmg-prefix (if parmmg is enabled) can be more maintainable than updating the `getall` mmg commit each time?

Comment (prj-) [reply 3rdparty/parmmg/Makefile:36]: Regarding your first comment: that's alright, users who download stuff via `--enable-download` usually don't mind if it's not the latest version, otherwise they can still compile their favorite libraries themselves. For your second comment: it's maybe cleaner, but then you have to go through (at least) two different code paths for Mmg: with or without ParMmg, so it's not that easy to maintain as well.

To sum up, could you please revert this change and bump the Mmg commit in `getall`? I can fix this in your branch directly if I have write access. Thanks!

**Comment by prj-** [comment]
Thanks, I'll check the branch on Windows now to see if I can merge it.
